src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/innerassignment/InputInnerAssignmentSwitchAndSwitchExpression.java

```diff
@@ -0,0 +1,51 @@
+/*
+InnerAssignment
+
+
+*/
+package com.puppycrawl.tools.checkstyle.checks.coding.innerassignment;
+
+//non-compiled with javac: Compilable with Java14
+public class InputInnerAssignmentSwitchAndSwitchExpression {
+
+    public void test1(int mode) {
+        int x = 0;
+        switch(mode) {
+            case 2 -> {
+                x = 2;
+            }
+            case 1 -> x = 1;
+        }
+    }
+
+    public void test2(int mode) {
+        int x = 0;
+        x = switch (mode) {
+            case 2 -> {
+                yield x = 2; // violation
+            }
+            case 1 -> x = 1;  // violation
+            default -> throw new UnsupportedOperationException();
```

Comment:
Please add inner assignment to default branch, and create another method with the same pattern, but with switch labeled statement groups. Also, add an example with multiple case constants in one or more labels.

Reply:
done and updated documentation to add examples with switch